### PR TITLE
MoreUtils.pm: implemented slice_missing

### PR DESCRIPTION
**Pull request by 1100Zimmermann**
added the slice_missing functionality according to the
given tests and some documentation of it.

Signed-off-by: Christoph Zimmermann <christophemzim@web.de>

### CODE REVIEW

File: lib/Hash/MoreUtils.pm [hunk 1]
```diff
@@ -55,6 +55,13 @@ hashref.
 
 If no C<LIST> is given, all keys are assumed as C<LIST>.
 
+=head2 C<slice_missing> HASHREF[, LIST]
+
+Returns a hash containing the (key => undef) pair for every
+LIST (as key) item not in hashref.
+
+If no C<LIST> is given, the returned map is empty.
```
Comment (rehsack) [inline lib/Hash/MoreUtils.pm]: Please explain in PD why the returned list is empty.

File: lib/Hash/MoreUtils.pm [hunk 2]
```diff
@@ -126,6 +138,12 @@ hashref.
 
 If no C<MAP> is given, all keys of C<HASHREF> are assumed mapped to themselves.
 
+=head2 C<slice_missing_map> HASHREF[, MAP]
+
+As C<slice_missing> but only checks for missing keys from MAP in C<HASHREF>.
+
+If no C<MAP> is given, C<slice_missing> will be used only on C<HASHREF>.
```
Comment (rehsack) [inline lib/Hash/MoreUtils.pm]: 1. slice_missing_map
2. checks for missing keys and map to ... from ...
3. returns ... - empty result when MAP is omitted

File: t/01-hash.t [hunk 3]
```diff
@@ -166,6 +166,23 @@ is_deeply(
...
+                f => "F"
+            )
+        )
+    },
+    {F => undef},
+    "slice_missing_map with given map",
+);
+
+is_deeply({slice_missing_map(\%h)}, {}, "slice_missing_map using slice_missing",);
+
```
Comment (rehsack) [inline t/01-hash.t]: Ideally add a test for empty map with empty result set.

File: lib/Hash/MoreUtils.pm [hunk 4]
```diff
@@ -57,11 +57,18 @@ If no C<LIST> is given, all keys are assumed as C<LIST>.
...
-Returns a hash containing the (key => undef) pair for every
-LIST (as key) item not in hashref.
+Returns a C<HASHREF> containing the (key => undef) pair for every
+C<LIST> (as key) item not in hashref.
 
 If no C<LIST> is given, the returned map is empty.
 
+=head2 C<slice_notdef> HASHREF[, LIST]
+
+As notdef understood as undef this function returns a
```
Comment (rehsack) [inline lib/Hash/MoreUtils.pm]: In case you're inconvenient with slice_notdef and prefer slice_undef, feel free to rename the function to make yourself more convenient.

File: lib/Hash/MoreUtils.pm [hunk 5]
```diff
@@ -57,11 +57,18 @@ If no C<LIST> is given, all keys are assumed as C<LIST>.
...
+C<HASHREF> containing the (key -> undef) pair for every
+C<LIST> (as key) item that is in the C<HASHREF> but with
+an undefined value.
+
```
Comment (rehsack) [inline lib/Hash/MoreUtils.pm]: Missing return description

File: lib/Hash/MoreUtils.pm [hunk 6]
```diff
@@ -144,6 +158,12 @@ As C<slice_missing> but only checks for missing keys from MAP in C<HASHREF>.
 
 If no C<MAP> is given, C<slice_missing> will be used only on C<HASHREF>.
 
+=head2 C<slice_notdef_map> HASHREF[, MAP]
+
+As C<slice_notdef> but only checks for undefined keys from MAP in C<HASHREF>.
+
+If no C<MAP> is given, C<slice_notdef> will be used only on C<HASHREF>.
```
Comment (rehsack) [inline lib/Hash/MoreUtils.pm]: which implies ...

File: t/01-hash.t [hunk 7]
```diff
@@ -183,6 +187,23 @@ is_deeply(
...
+                c => "C"
+            )
+        )
+    },
+    {C => undef},
+    "slice_notdef_map with given map",
+);
+
+is_deeply({slice_notdef_map(\%h)}, {}, "slice_notdef_map using slice_notdef",);
+
```
Comment (rehsack) [inline t/01-hash.t]: empty map missing

**Comment by rehsack** [comment]
Merged
